services/rollout-service/pkg/service/service.go

```diff
@@ -45,10 +48,29 @@ var (
 )
 
 type ArgoEventProcessor interface {
-	ProcessArgoEvent(ctx context.Context, ev ArgoEvent)
+	ProcessArgoEvent(ctx context.Context, ev ArgoEvent) *ArgoEvent
 }
 
-func ConsumeEvents(ctx context.Context, appClient SimplifiedApplicationServiceClient, dispatcher *Dispatcher, hlth *setup.HealthReporter, a *argo.ArgoAppProcessor, ddMetrics statsd.ClientInterface) error {
+type ConsumeEventsParameters struct {
+	AppClient           SimplifiedApplicationServiceClient
+	Dispatcher          *Dispatcher
+	HealthReporter      *setup.HealthReporter
+	ArgoAppProcessor    *argo.ArgoAppProcessor
+	DDMetrics           statsd.ClientInterface
+	DBHandler           *db.DBHandler
+	PersistArgoEvents   bool
+	ArgoEventsBatchSize int
+}
+
+func ConsumeEvents(ctx context.Context, params *ConsumeEventsParameters) error {
+	hlth := params.HealthReporter
+	appClient := params.AppClient
+	dispatcher := params.Dispatcher
+	a := params.ArgoAppProcessor
+	ddMetrics := params.DDMetrics
+	handler := params.DBHandler
```

Comment:
Not sure how difficult to implement this but I think the better way is to define a struct Called `EventsConsumer` with these fields. Then `ConsumeEvents` should be a method of that struct, Like this:
```
func (e *EventConsumer) ConsumeEvents(ctx context.Context) error {
...
```
and then use these params whenever needed like this: `e.AppClient`